### PR TITLE
Upgrade to Apache Calcite 1.23

### PR DESCRIPTION
**Pull request by eolivelli**
Upgrade to Apache Calcite 1.23
As you can see we had to change one test case because Calcite looks smarter

### CODE REVIEW

**Comment by eolivelli** [comment]
Calcite 1.23 vote has been closed, waiting for artifacts to be available on Maven Central.

@diegosalvi  @amitvc  please take a look and merge as soon as Travis is green.
This is the last important patch before 0.16.0 release

**Comment by eolivelli** [comment]
We have a compatibility test failing:
```
ROR] test(herddb.upgrade.RunHerdDB070Test)  Time elapsed: 6.957 s  <<< ERROR!
herddb.model.StatementExecutionException: herddb.core.HerdDBInternalException: This datascanner is not rewindable
```
We should have those code paths tested in "core" project !

File: herddb-core/src/test/java/herddb/sql/SimplerPlannerTest.java [hunk 1]
```diff
@@ -242,10 +242,13 @@ public void plannerBasicStatementsTest() throws Exception {
                     List<DataAccessor> results = scan.consume();
                     assertEquals(2, results.size());
                     assertEquals(3, results.get(0).getFieldNames().length);
-                    assertEquals(1234L, results.get(0).get(0));
+                    // see https://issues.apache.org/jira/browse/CALCITE-3998
```
Comment (amitvc) [inline herddb-core/src/test/java/herddb/sql/SimplerPlannerTest.java:245]: Interesting improvement by calcite. I read you correspondence on https://issues.apache.org/jira/browse/CALCITE-3998.

**Comment by eolivelli** [comment]
@amitvc there is a test that is failing.
so we need to investigate more.

There is also a problem with Maven Central, Calcite 1.23 is not available yet but the temporary repository that I used for testing has been dropped, because the VOTE has been closed.
I have sent an email to the dev@calcite.apache.org ML and they are fixing it.

As soon as they fix it I will debug the failing test case.

**Comment by eolivelli** [comment]
@amitvc Travis is passed, could you please help me merging this patch ?
Use "Squash and merge" and clean up the commit message, just leaving "Upgrade to Apache Calcite 1.23"

Thanks in advance

**Pull request merged**